feat(bedrock): add OpenAI-compatible service_tier parameter translation

Chesars opened the pull request:
## Relevant issues

Related to #17911 and #17810

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

## Type

🆕 New Feature

## Changes

### Summary

Adds OpenAI-compatible `service_tier` parameter support for Bedrock Converse API, translating it to Bedrock's native `serviceTier` format.

Currently, using the OpenAI-style `service_tier` parameter with Bedrock models throws `UnsupportedParamsError`:

```python
# This fails with UnsupportedParamsError
litellm.completion(
    model="bedrock/converse/anthropic.claude-3-sonnet-20240229-v1:0",
    messages=[{"role": "user", "content": "Hi"}],
    service_tier="priority"
)
```

### Solution

1. Added `service_tier` to `supported_params` in `get_supported_openai_params()`
2. Added translation in `map_openai_params()` that converts:
   - `service_tier="priority"` → `serviceTier={"type": "priority"}`
   - `service_tier="default"` → `serviceTier={"type": "default"}`
   - `service_tier="flex"` → `serviceTier={"type": "flex"}`
   - `service_tier="auto"` → `serviceTier={"type": "default"}` (Bedrock doesn't support "auto")

### Usage

```python
# Now works with OpenAI-style parameter
litellm.completion(
    model="bedrock/converse/anthropic.claude-3-sonnet-20240229-v1:0",
    messages=[{"role": "user", "content": "Hi"}],
    service_tier="priority"  # OpenAI format, automatically translated
)
```

### Tests

Added 5 new tests in `tests/test_litellm/llms/bedrock/chat/test_service_tier.py`:
- `test_service_tier_in_supported_openai_params`
- `test_map_openai_service_tier_priority`
- `test_map_openai_service_tier_default`
- `test_map_openai_service_tier_flex`
- `test_map_openai_service_tier_auto_maps_to_default`

12 tests passing:
```
======================== 12 passed in 0.04s ========================
```

Chesars committed:
feat(bedrock): add OpenAI-compatible service_tier parameter translation

Translates OpenAI's service_tier parameter (string) to Bedrock's
serviceTier format (object with type field).

krrishdholakia commented on litellm/llms/bedrock/chat/converse_transformation.py:
@Chesars do we need to update docs for this? 

not sure if serviceTier is a documented parameter. If not - can you update bedrock.md with this documented? so people know this exists and how to use it

Chesars replied:
Yeah, let's add docs.

Chesars committed:
docs(bedrock): add OpenAI-compatible service_tier parameter documentation

Document the automatic translation from OpenAI-style service_tier
parameter to Bedrock's native serviceTier format.

kuldeepjain commented:
@Chesars @krrishdholakia Could you please get this merged, if it's ready to go?

Chesars committed:
feat(bedrock): add service_tier to response when present

According to OpenAI's API documentation, when service_tier is sent in the
request, it should be returned in the response. This commit implements
this behavior for Bedrock Converse API to maintain compatibility with
OpenAI's API.

Changes:
- Added serviceTier field to ConverseResponseBlock type definition
- Moved ServiceTierBlock definition before ConverseResponseBlock to fix
  type reference order
- Added response transformation to map Bedrock serviceTier (object) to
  OpenAI service_tier (string format)
- Added 4 new tests for response transformation with service_tier

The service_tier is only added to the response when present in Bedrock's
response, maintaining backward compatibility.
